### PR TITLE
*: Support sleep function

### PR DESCRIPTION
**Pull request by zimulala**

### CODE REVIEW

File: evaluator/builtin_other.go [hunk 1]
```diff
@@ -25,6 +26,28 @@ import (
 	"github.com/pingcap/tidb/util/types"
 )
 
+// See http://dev.mysql.com/doc/refman/5.7/en/miscellaneous-functions.html#function_sleep
+func builtinSleep(args []types.Datum, _ context.Context) (d types.Datum, err error) {
+	if args[0].IsNull() {
```
Comment (shenli) [inline evaluator/builtin_other.go]: MySQL will treat null or negative numbers as zero.
mysql> select sleep(null);
+-------------+
| sleep(null) |
+-------------+
|           0 |
+-------------+
1 row in set (0.00 sec)

mysql> select sleep(-1);
+-----------+
| sleep(-1) |
+-----------+
|         0 |
+-----------+
1 row in set (0.00 sec)

**Comment by shenli** [comment]
Rest LGTM

**Comment by hanfei19910905** [comment]
LGTM

**Comment by shenli** [comment]
LGTM

**Comment by coocood** [comment]
LGTM

File: evaluator/builtin_other.go (hunk 1 above)
Comment (zimulala) [inline evaluator/builtin_other.go]: In the MySQL 5.7 doc has the following description: If the argument is NULL or negative, SLEEP() produces a warning, or an error in strict SQL mode.
I'll add SQL mode judgement.

**Pull request merged**